internal/nginx/config/validation/http_filters.go

```diff
@@ -0,0 +1,39 @@
+package validation
+
+// HTTPRedirectValidator validates values for a redirect, which in NGINX is done with the return directive.
+// For example, return 302 "https://example.com:8080";
+type HTTPRedirectValidator struct{}
+
+var supportedRedirectSchemes = map[string]struct{}{
+	"http":  {},
+	"https": {},
+}
+
+// ValidateRedirectScheme validates a scheme to be used in the return directive for a redirect.
+// NGINX rules are not restrictive, but it is easier to validate just for two allowed values http and https,
+// dictated by the Gateway API spec.
+func (HTTPRedirectValidator) ValidateRedirectScheme(scheme string) (valid bool, supportedValues []string) {
+	return validateInSupportedValues(scheme, supportedRedirectSchemes)
+}
+
+func (HTTPRedirectValidator) ValidateRedirectHostname(hostname string) error {
+	return validateEscapedStringNoVarExpansion(hostname, "host", "example.com")
+}
+
+func (HTTPRedirectValidator) ValidateRedirectPort(_ int32) error {
```

Comment:
this field ends up in the data plane configuration. I'd like to have check it and leave it for the data plane to decide if it is valid.

Reply:
Ok, but since we aren't checking anything for NGINX, I wonder if it makes sense to include this method now. Should we wait until we have a data plane or implementation that requires this check?

Reply:
I'd rather keep the list of fields that propagate to data plane config this way, for visibility and enforcement. For example, let's imagine we add another data plane implementation. If we don't have this check,  but let's say we only have a comment, the data plane will not implement this check, unless we remember about the comment. If we have the check, the data plane will _have_ to implement the check.

Reply:
Does every field in HTTPRoute have a corresponding validate method on the interface?

Reply:
nope. a lot of fields don't, because they don't affect configuration directly. For example, matching type, filter type, fields related to external references

Reply:
Ok. I feel like this is YAGNI, but I also see the benefits of leaving it as is and I don't want to hold up the PR.